qiskit/algorithms/minimum_eigen_solvers/adaptvqe2.py

```diff
@@ -0,0 +1,375 @@
+        sampler = CircuitSampler(self.quantum_instance)
+        for exc in self._excitation_pool:
+            # add next excitation to ansatz
+            print(exc)
+            print(len(self._excitation_list))
+            self._tmp_ansatz.operators = self._excitation_list + [exc]
+            # the ansatz needs to be decomposed for the gradient to work
+            self.ansatz = self._tmp_ansatz.decompose()
+            print(self.ansatz)
+            param_sets = list(self.ansatz.parameters)
+            print(param_sets)
+            # zip will only iterate the length of the shorter list
+            theta1 = dict(zip(self.ansatz.parameters, theta))
+            op, expectation = self.construct_expectation(theta1, self._operator, return_expectation=True)
+            # compute gradient
+            state_grad = self.gradient.convert(operator=op, params=param_sets)
+            # Assign the parameters and evaluate the gradient
+            value_dict = {param_sets[-1]: 0.0}
+            print(value_dict)
+            state_grad_result = sampler.convert(state_grad, params=value_dict).eval()
+            print(state_grad_result)
+            logger.info("Gradient computed : %s", str(state_grad_result))
+            res.append((np.abs(state_grad_result[-1]), exc))
+        return res, expectation
+
+    @staticmethod
+    def _check_cyclicity(indices: List[int]) -> bool:
+        """
+        Auxiliary function to check for cycles in the indices of the selected excitations.
+
+        Args:
+            indices: the list of chosen gradient indices.
+        Returns:
+            Whether repeating sequences of indices have been detected.
+        """
+        cycle_regex = re.compile(r"(\b.+ .+\b)( \b\1\b)+")
+        # reg-ex explanation:
+        # 1. (\b.+ .+\b) will match at least two numbers and try to match as many as possible. The
+        #    word boundaries in the beginning and end ensure that now numbers are split into digits.
+        # 2. the match of this part is placed into capture group 1
+        # 3. ( \b\1\b)+ will match a space followed by the contents of capture group 1 (again
+        #    delimited by word boundaries to avoid separation into digits).
+        # -> this results in any sequence of at least two numbers being detected
+        match = cycle_regex.search(" ".join(map(str, indices)))
+        logger.debug("Cycle detected: %s", match)
+        # Additionally we also need to check whether the last two numbers are identical, because the
+        # reg-ex above will only find cycles of at least two consecutive numbers.
+        # It is sufficient to assert that the last two numbers are different due to the iterative
+        # nature of the algorithm.
+        return match is not None or (len(indices) > 1 and indices[-2] == indices[-1])
+
+    def solve(
+        self,
+        aux_operators: Optional[ListOrDictType[PauliSumOp]] = None,
+        #operator: OperatorBase = None
+    ) -> "AdaptVQEResult":
```

Comment:
When I remove quotes it shows: `"AdaptVQEResult" is not defined`

Reply:
Yes, that's because its a forward reference to something defined later in the same module. You can add this upfront in the module (for 3.7 and above, but since 3.6 is no longer supported it works across the supported versions)
>from __future__ import annotations